### PR TITLE
#972 TProcessHelper, TSignalsDispatcher, and related classes & functions

### PR DESCRIPTION
**Pull request by belisoful**
- TProcessHelper for process related functions like forking (pcntl_fork), isSystemWindows(), isSystemMacOS, and isSystemLinux.
- TSignalsDispatcher for raising signal related global events from signals for multiple handlers per signal, alarm and disarm interrupt callbacks at a specific time (1 second precision), and for callbacks per child processes (proc_open, pcntl_fork, etc) when they end (or "stop and start"?).
- ISingleton is a new interface for application singleton objects
- TShellWriter is updated to use the TProcessHelper::isSystemWindows function 
- -TComponent hasMethod of global event handlers is true for methods that do actually exist.  As a global event it exists always, but not the method.
- TEventSubscription example is updated to reflect TSignalsDispatcher and TSignalParameter
- TApplicationSignals behavior for configuration and attaching of TSignalsDispatcher
- TCaptureForkLog behavior for capturing the logs of forked child processes in the main log.  Effectively a ForkLogRouter but registering as a behavior on TApplication.
- TForkable behavior for attaching the owner's ::fxPrepareForFork() and ::fxRestoreAfterFork() handlers to their respective global events. 
- TGlobalClassAware behavior for attaching the owner's ::fxAttachClassBehavior() and ::fxDetachClassBehavior() to their respective global events so they change with global class changes without listening.
- TProcessWindowsPriority specifies the numeric priorities that windows uses for each respective priority level.
- TProcessWindowsPriorityName specifies the text priorities that windows uses for each respective priority level.

### CODE REVIEW

**Comment by belisoful** [comment]
This took a while to ensure proper function and consideration.  This is ready for review and merge.

File: framework/classes.php [hunk 1]
```diff
@@ -276,7 +277,11 @@
 'TModule' => 'Prado\TModule',
 'TPropertyValue' => 'Prado\TPropertyValue',
 'TService' => 'Prado\TService',
+'TApplicationSignals' => 'Prado\Util\Behaviors\TApplicationSignals',
 'TBehaviorParameterLoader' => 'Prado\Util\Behaviors\TBehaviorParameterLoader',
+'TCaptureForkLog' => 'Prado\Util\Behaviors\TCaptureForkLog',
+'Forkable' => 'Prado\Util\Behaviors\Forkable',
```
Comment (ctrlaltca) [inline framework/classes.php:283]: Typo: TForkable

File: framework/classes.php [hunk 2]
```diff
@@ -332,6 +340,8 @@
 'TRpcClientRequestException' => 'Prado\Util\TRpcClientRequestException',
 'TRpcClientResponseException' => 'Prado\Util\TRpcClientResponseException',
 'TRpcClientTypesEnumerable' => 'Prado\Util\TRpcClientTypesEnumerable',
+'TSignalsDispatcher' => 'Prado\Util\TSignalsDispatcher',
+'TSignalsParameter' => 'Prado\Util\TSignalsParameter',
```
Comment (ctrlaltca) [inline framework/classes.php:344]: Typo: TSignalParameter

**Review by ctrlaltca — CHANGES_REQUESTED** [review]
Sorry for the delay, the PR looks fine with the exception of a couple of typos.

**Comment by belisoful** [comment]
No issue with the delay.  It's given me a little break that I needed.  I'll get on those typos asap

**Comment by belisoful** [comment]
TSignalsParameter was the first pass name of the object.  further revision made is singular and the class file didn't get updated.  good catch.

**Pull request merged**